src/validation/trips/index.js

```diff
@@ -110,52 +110,42 @@ export const validateTrips = async (req, res, next) => {
-    const origin = locations.getCountryByShort(country.toUpperCase());
-    const { suggestions, error, names } = helper.getCityData(origin.states, city);
-    if (error) {
-      return resSend(res, 400, error, { suggestions });
-    }
-    if (!names.find(place => place === city.toLocaleLowerCase())) {
-      return resSend(res, 400, `we dint find ${city} in  ${origin.name}`);
-    }
-    // same destination validation
-    if ((req.body.city || req.body.trip.city)
-    && (req.body.city.toLocaleLowerCase() || req.request.city)
-    === (req.body.trip.city.toLocaleLowerCase() || req.trip.city)) {
-      throw new Error('trips to the same city not allowed');
-    }
+  const { country, city, trip, returnTime } = req.body;
+  const origin = locations.getCountryByShort(country.toUpperCase());
+  const { suggestions, error, names } = helper.getCityData(origin.states, city);
+  if (error) {
+    return resSend(res, 400, error, { suggestions });
+  }
+  if (!names.find(place => place === city.toLocaleLowerCase())) {
+    return resSend(res, 400, `we dint find ${city} in  ${origin.name}`);
+  }
+  // same destination validation
+  if ((city || trip.city)
+    && (city.toLocaleLowerCase() || req.request.city)
+    === (trip.city.toLocaleLowerCase() || req.trip.city)) {
+    return resSend(res, 400, 'trips to the same city not allowed');
+  }
 
-    if ((req.body.returnTime || req.body.trip.departureTime)
-    && new Date(req.body.returnTime || req.request.returnTime)
-    <= new Date(req.body.trip.departureTime || req.request.returnTime)) {
-      throw new Error('returnTime can not be less than OR equal to departureTime');
-    }
-    next();
-  } catch (err) {
-    return resSend(res, 400, err.message);
+  if ((returnTime || trip.departureTime)
+    && (new Date(returnTime || req.request.returnTime)
+    <= new Date(trip.departureTime || req.trip.departureTime))) {
+    return resSend(res, 400, 'returnTime can not be less than OR equal to departureTime');
   }
+  next();
 };
 
 export const validateTripsData = async (req, res, next) => {
-  try {
-    const { country, city, departureTime } = req.body.trip;
-    // check date
-    if (departureTime && !departureTime.match(/[12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])/)) {
-      return resSend(res, 400, 'date fomart is invalid try: YYYY-M-D');
-    }
-    const destination = locations.getCountryByShort(country.toUpperCase());
-    const { suggestions, error, names } = helper.getCityData(destination.states, city);
-    if (error) {
-      return resSend(res, 400, error, { suggestions });
-    }
-
-    if (!names.find(place => place === city.toLocaleLowerCase())) {
-      return resSend(res, 400, `we dint find ${city} in  ${destination.name}`);
-    }
-    next();
-  } catch (err) {
-    return resSend(res, 400, err.message);
+  const { country, city, departureTime } = req.body.trip;
+  if (departureTime && !departureTime.match(/[12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01])/)) {
```

Comment:
Can you put this inside a validator, if there is no one already to handle date format? In that way, all references to date formats will use the validator.